### DISCUSSION TITLE
Create a set of primitive components for web applications

### DISCUSSION THREAD

**Discussion by andreaSimonePorceddu** [morfeojs/morfeo#64] [Ideas]
**Is your feature request related to a problem? Please describe.**
Create a set of ready to use primitive components as a new package (@morfeo/primitives). At now the morfeo eco-system doesn't provide a way to quick start with the UI development. This feature should reduce the amount of boilerplate needed to start to develop with Morfeo.

**Describe the solution you'd like**
I'm not sure about what approach could be the best for this feature yet. A solution that can work fine on all frameworks could be to develop the set of primitives as web components by using a library like stenciljs (https://stenciljs.com/).

**Describe alternatives you've considered**
Another approach could be to develop a common set of component styles by using the theme specs of morfeo and then develop separated solutions for each of the main frameworks and libraries:

- primitives-react
- primitives-web-components
- primitives-vue
- primitives-svelte
- primitives-angular

**Comment by mauroerta** [comment]
I think the best way could be use to `stenciljs` (or another web-components library) in order to make only one package that can be used in every framework or environment like React, Angular or svelte

**Reply by andreaSimonePorceddu** [reply]
It depends also on what kind of limits a web-component approach could have. I think we have to evaluate the best solution by experimenting with the first one and trying to understand if it can match all needs of every specific framework.

**Reply by mauroerta** [reply]
Can you try with `stenciljs` and make a POC inside the sandboxes?
